### PR TITLE
More skeletal zombie upgrades

### PR DESCRIPTION
**Pull request by SirPendrak**
#### Summary
Content "Three new skeletal zombe evolutions"

#### Purpose of change

More variety and endgame enemies

#### Describe the solution

Skeletal slasher - can evolve from skeletal zombie. Similar armor, a tiny bit faster, better melee skill and token 1 dodge. "Attack cost" makes it attack once per turn despite speed lower than 100. Can smell. Has good ammount of cut damage to its attack.

Skeletal executioneer - slasher upgrade. Looses grab and bite. Has very good melee skill, a lot of cut DMG and tries to decapitate you with powerfull "cut_throat" attack. Otherwise has some minor stat upgrades.

Skeletal reaper - the blob got serious. Extreme melee skill and cutting damage. From time to time can perform armor-penetrating stab. Larger, faster, stronger. Probably the longest zombie evolution chain so far. Looses smell due to its nostrills being overgrown with bone. Can push small vehicles.

#### Describe alternatives you've considered

Fighting Skeletal juggernaut again

#### Testing

Despite big numbers they are not that deadly for survivor that has at least some preparation. Cutting damage is easiest to negate with armor. 

Tested that zombie masters upgrade skeletons into slashers at about 12% rate. 

Fought skeletal slasher. Endgame armor can make you impervious to their attacks. They loose fight to a standard Mi-go, leaving it with about half HP.

Fought skeletal executioneer. As long as you have good armor and can block or parry, youre fine. Otherwise you loose your head fast. It looses fight with Mi-go guard, leaving it at 10% HP left. 

Fought skeletal reaper. Their speed makes them more annoyig and overwhelming. As above, parry and armor should keep player safe, but melee skill of 9 makes pure dodge builds risky. It looses to a Mi-go myrmidon, but barerly, with Mi-go having about 5% HP left. In bigger numbers, or when they attack with some grabbers or similar they pose true endgame threat if ignored. 

#### Additional context

I am not native english speaker. Any points about zombie descrieptions are welcome

### CODE REVIEW

File: data/json/monsters/zed_skeletal.json [hunk 1]
```diff
@@ -112,5 +112,112 @@
...
+    "melee_damage": [ { "damage_type": "cut", "amount": 25 } ],
+    "dodge": 1,
+    "bleed_rate": 0,
+    "scents_tracked": [ "sc_human", "sc_fetid" ],
+    "special_attacks": [ { "id": "cut_throat", "damage_max_instance": [ { "damage_type": "cut", "amount": 40 } ], "cooldown": 15 } ],
+    "upgrades": { "half_life": 32, "into": "mon_skeleton_reaper" },
+    "fungalize_into": "mon_skeleton_brute_fungus",
+    "burn_into": "mon_zombie_fiend",
+    "armor": { "bash": 10, "cut": 35, "stab": 30, "acid": 4, "bullet": 25, "electric": 5 },
+    "flags": [
```
Comment (Karol1223) [inline data/json/monsters/zed_skeletal.json:169]: Handle this via ``extend``/``delete``, not manually defining flags.

File: data/json/monsters/zed_skeletal.json [hunk 2]
```diff
@@ -112,5 +112,112 @@
...
+    "bleed_rate": 0,
+    "special_attacks": [
+      { "id": "cut_throat", "damage_max_instance": [ { "damage_type": "cut", "amount": 45 } ], "cooldown": 15 },
+      { "id": "impale", "damage_max_instance": [ { "damage_type": "stab", "amount": 40 } ], "cooldown": 20 }
+    ],
+    "upgrades": false,
+    "fungalize_into": "mon_skeleton_brute_fungus",
+    "burn_into": "mon_zombie_fiend",
+    "armor": { "bash": 15, "cut": 35, "stab": 35, "acid": 4, "bullet": 30, "electric": 5 },
+    "flags": [
```
Comment (Karol1223) [inline data/json/monsters/zed_skeletal.json:208]: As above

**Review by Karol1223 — CHANGES_REQUESTED** [review]

**Comment by RAldrich** [comment]
Why is the slasher able to smell? None of the other skeletons do so, and I thought we were reserving that ability for monsters that evolved from things like dogs (or the stalker series of zombies). I think it would make sense to remove that ability.

**Comment by Karol1223** [comment]
> Why is the slasher able to smell? None of the other skeletons do so, and I thought we were reserving that ability for monsters that evolved from things like dogs (or the stalker series of zombies). I think it would make sense to remove that ability.

It's been significantly more restricted than it used to be, but smell is pretty common around the brute-tier of enemies still, including both skeletal brutes and juggernauts.

File: data/json/monsters/zed_skeletal.json [hunk 3]
```diff
@@ -112,5 +112,88 @@
     "fungalize_into": "mon_skeleton_hulk_fungus",
     "armor": { "bash": 20, "cut": 45, "bullet": 36, "electric": 8 },
     "extend": { "weakpoint_sets": [ "wps_bone_armor" ], "families": [ "prof_wp_skeleton" ] }
+  },
+  {
+    "id": "mon_skeleton_slasher",
+    "type": "MONSTER",
+    "name": { "str": "skeletal slasher" },
+    "description": "The fingers of this bone-clad zombie have turned into ossified blades the size of large knives. Most of the soft tissue on its head is gone, exposing a significantly thickened skull.",
```
Comment (Karol1223) [inline data/json/monsters/zed_skeletal.json:120]:
```suggestion
    "description": "The fingers of this bone-clad zombie have turned into ossified blades the size of large knives.  Most of the soft tissue on its head is gone, exposing a significantly thickened skull.",
```

File: data/json/monsters/zed_skeletal.json [hunk 4]
```diff
@@ -112,5 +112,88 @@
...
+    ],
+    "upgrades": { "half_life": 32, "into": "mon_skeleton_executioner" },
+    "armor": { "bash": 5, "cut": 30, "stab": 25, "acid": 3, "bullet": 20, "electric": 3 },
+    "extend": { "flags": [ "SMELLS" ] }
+  },
+  {
+    "id": "mon_skeleton_executioner",
+    "type": "MONSTER",
+    "name": { "str": "skeletal executioner" },
+    "description": "The hands and forearms of this tall, bone-clad zombie have transformed into sword-like ossified blades. Its jaw is partially fused with the bone mask covering its face, leaving sinister-looking gaps. Its movements are stiff but precise.",
```
Comment (Karol1223) [inline data/json/monsters/zed_skeletal.json:151]:
```suggestion
    "description": "The hands and forearms of this tall, bone-clad zombie have transformed into sword-like ossified blades.  Its jaw is partially fused with the bone mask covering its face, leaving sinister-looking gaps.  Its movements are stiff but precise.",
```

File: data/json/monsters/zed_skeletal.json [hunk 5]
```diff
@@ -112,5 +112,88 @@
...
+    "burn_into": "mon_zombie_fiend",
+    "armor": { "bash": 10, "cut": 35, "stab": 30, "acid": 4, "bullet": 25, "electric": 5 },
+    "extend": { "flags": [ "SMELLS" ] },
+    "delete": { "flags": [ "GRABS" ] }
+  },
+  {
+    "id": "mon_skeleton_reaper",
+    "type": "MONSTER",
+    "name": { "str": "skeletal reaper" },
+    "description": "This large, bone-armored zombie has scythe-like blades in place of its hands, resembling a praying mantis. Its face is covered by a thick, ossified mask, with its only remaining facial features being eyes that ooze black goo. Its movements are filled with unnatural rage.",
```
Comment (Karol1223) [inline data/json/monsters/zed_skeletal.json:176]:
```suggestion
    "description": "This large, bone-armored zombie has scythe-like blades in place of its hands, resembling a praying mantis.  Its face is covered by a thick, ossified mask, with its only remaining facial features being eyes that ooze black goo.  Its movements are filled with unnatural rage.",
```

**Review by Karol1223 — CHANGES_REQUESTED** [review]
You gotta leave double spaces after periods

**Pull request merged**